### PR TITLE
Patch visualizer redesign

### PR DESCRIPTION
**Pull request by boatbomber**
The patch visualizer component has a nicer tree design that is more readable while also more compact.

The confirming page has been slimmed down to just the necessary info:

image image

The info for the latest patch has moved from text along the bottom into an icon in the top:

*(No warnings)*
image image
*(With warnings)*
image image

Clicking on the patch details icon will now open a detailed pop up, so that users can view details without resizing their main window:

*(No warnings)*
image image
*(With warnings)*
image image

### CODE REVIEW

**Review by Dekkonot — COMMENTED** [review]
Have we considered the performance impact of using editable images? I imagine it's relatively low, but I also have a pretty good computer so I'm not a good judge.

**Comment by boatbomber** [comment]
Performance hit is negligible once generated (it's just an image texture, so only mem cost), but there's a small cost up front when building them. It's minimal though.

To help mitigate it, I've added caching for the classicon pixel data so that we no longer need to get it from AssetService every time.

File: plugin/src/App/Components/ClassIcon.lua [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+local e = Roact.createElement
+
+local EditableImage = require(Plugin.App.Components.EditableImage)
+
+local function getRecoloredClassIcon(className, color)
+	-- HACK: DataModel is currently missing an icon so we redirect it
+	if className == "DataModel" then
+		className = "Class"
+	end
+
```
[the remark this answers is not in the document]
Comment (boatbomber) [reply plugin/src/App/Components/ClassIcon.lua]: It's not resolved for me, but I have custom icons so perhaps it's just bugged on my machine. Removing it and trusting that even if it is broken, it'll be resolved soon enough.

**Review by Dekkonot — APPROVED** [review]
Looks good to me.

**Pull request merged**